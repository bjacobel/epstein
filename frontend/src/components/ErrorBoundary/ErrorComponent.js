import React from 'react';

import { SHOW_DEV_TOOLS } from '../../constants';
import { errorHeader } from './style.css';

export default ({ error }) => {
  return (
    <>
      <h3 className={errorHeader}>Something went wrong.</h3>
      {SHOW_DEV_TOOLS && error && error.stack && <pre>{error.stack}</pre>}
    </>
  );
};
