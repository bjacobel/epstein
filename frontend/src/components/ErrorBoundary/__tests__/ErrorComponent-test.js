import React from 'react';
import { shallow } from 'enzyme';

import ErrorComponent from '../ErrorComponent';

describe('error rendering component', () => {
  describe('in dev mode', () => {
    it('matches snapshot without an error', () => {
      expect(shallow(<ErrorComponent />)).toMatchSnapshot();
    });

    it('matches snapshot with an error', () => {
      expect(
        shallow(<ErrorComponent error={new Error('idk lol')} />)
          .find('pre')
          .text(),
      ).not.toBeUndefined();
    });
  });

  describe('in prod mode', () => {
    it('does not print stack even with an error', () => {
      jest.resetModules();
      jest.doMock('../../../constants', () => ({
        SHOW_DEV_TOOLS: false,
      }));
      // eslint-disable-next-line global-require
      const ScopedErrorComponent = require('../ErrorComponent').default;

      expect(
        shallow(<ScopedErrorComponent error={new Error('idk lol')} />).find('pre').length,
      ).toEqual(0);
    });
  });
});
