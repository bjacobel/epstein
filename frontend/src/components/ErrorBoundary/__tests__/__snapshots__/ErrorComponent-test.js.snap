// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`error rendering component in dev mode matches snapshot without an error 1`] = `
<Fragment>
  <h3
    className="errorHeader"
  >
    Something went wrong.
  </h3>
</Fragment>
`;
